Solved issue 3392. (#3430)
Codecademy/docs: content/lua/concepts/mathematical-library/terms/exp/exp.md

---
Title: 'Exp'

> Description: 'Exp is method used to calculate the exponential value of a number.'

Reviewer: ```suggestion
Description: 'Used to calculate the exponential value of a number.'
```

Subjects:
  - 'Computer Science'
  - 'Lua'
  - 'Game Development'
Tags:
  - 'Arithmetic'
  - 'Functions'
  - 'Math'
  - 'Numbers'
CatalogContent:
  - 'learn-lua'
  - 'paths/computer-science'
---

> In Lua, the **math.exp()** function is a standard function that is part of the Lua 'math' library. The math.exp() function is used to calculate the exponential value of a number, i.e, the exponential constant 'e' (which is roughly equal to 2.718281828459045...) raised to a given value. Exponents are a very common part of mathematical operations across a wide variety of fields and disciplines such as Physics, engineering, probability and statistics, and even machine learning! This means that the math.exp() function is a very useful and widely applicable tool for Lua developers.

Reviewer: ```suggestion
In Lua, the **`math.exp()`** function is a standard function that is part of the Lua `math` library. The `math.exp()` function is used to calculate the exponential value of a number, i.e, the exponential constant 'e' (which is roughly equal to 2.718281828459045...) raised to a given value. Exponents are a very common part of mathematical operations across a wide variety of fields and disciplines such as physics, engineering, probability and statistics, and even machine learning! This means that the `math.exp()` function is a very useful and widely applicable tool for Lua developers.
```

> As a side-note, the **math.log()** function in the math library returns the inverse of the math.exp() function.

Reviewer: ```suggestion
As a side-note, the **`math.log()`** function in the math library returns the inverse of the `math.exp()` function.
```

## Syntax

> As .exp() is a method that is a part of the standard Lua Math library, to use this function it must be called as 'math.exp()'

Reviewer: ```suggestion
As `.exp()` is a method that is a part of the standard Lua Math library, to use this function it must be called as `math.exp()`
```

> ```lua
> 
> math.exp(mycoolvalue) -- returns the result of e raised to the power of mycoolvalue
> 
> ```

Reviewer: ````suggestion
```pseudo
math.exp(mycoolvalue) -- returns the result of e raised to the power of mycoolvalue
```
````

> ## Example

Reviewer: There are some things that need to be changed with this example. The code needs to be changed so the code produces some output to the console. Then, instead of a comment in the code, the output needs to be shown in a ` ```shell` block after the example like this:

````
This example results in the following output:

```shell
Output from example
```
````
We also need an introductory sentence before the example summarizing what the example is doing.

---
Title: 'exp()'
Description: 'Used to calculate the exponential value of a number.'
Subjects:
  - 'Computer Science'

>   - 'Lua'

Reviewer 2: ```suggestion
```

  - 'Game Development'
Tags:
  - 'Arithmetic'
  - 'Functions'
  - 'Math'
  - 'Numbers'
CatalogContent:
  - 'learn-lua'
  - 'paths/computer-science'
---

> In Lua, the **`math.exp()`** function is a standard function that is part of the Lua `math` library. The `math.exp()` function is used to calculate the exponential value of a number, i.e, the exponential constant 'e' (which is roughly equal to 2.718281828459045...) raised to a given value. Exponents are a very common part of mathematical operations across a wide variety of fields and disciplines such as physics, engineering, probability and statistics, and even machine learning! This means that the `math.exp()` function is a very useful and widely applicable tool for Lua developers.

Reviewer 2: ```suggestion
In Lua, the **`math.exp()`** function is a standard function that is used to calculate the exponential value of a number, i.e, the exponential constant *'e'* (which is roughly equal to 2.718281828459045) raised to a given value. 

Exponents are a very common part of mathematical operations across a wide variety of fields and disciplines such as physics, engineering, probability and statistics, and even machine learning! This means that the `math.exp()` function is a very useful and widely applicable tool for Lua developers.
```

As a side-note, the **`math.log()`** function in the math library returns the inverse of the `math.exp()` function.

## Syntax

> As `.exp()` is a method that is a part of the standard Lua Math library, to use this function it must be called as `math.exp()`

Reviewer: ```suggestion
As `.exp()` is a method that is a part of the standard Lua `math` library, to use this function it must be called as `math.exp()`
```

> As `.exp()` is a method that is a part of the standard Lua `math` library, to use this function it must be called as `math.exp()`

Reviewer 2: ```suggestion
As `.exp()` is a method that is a part of the standard Lua `math` library, to use this function it must be called as `math.exp()`.
```

```pseudo

> math.exp(mycoolvalue) -- returns the result of e raised to the power of mycoolvalue

Reviewer 2: ```suggestion
math.exp(mycoolvalue) -- Returns the result of e raised to the power of mycoolvalue
```

```

## Example 1

> To find the exponential value of 0,

Reviewer 2: ```suggestion
To find the exponential value of `0`,
```

```lua
print(math.exp(0))
```

This results in the following output:

```shell
1.0
```

> This is because e raised to the power 0 (e^0) equals 1.

Reviewer 2: ```suggestion
```

## Example 2

> Similarly, to find the exponential value of 8,

Reviewer 2: ```suggestion
Similarly, to find the exponential value of `8`,
```

```lua
print(math.exp(8))
```

This results in the following output:

```shell
2980.9579870417
```

> This equals the value of e raised to the power 8 (e^8).

Reviewer 2: ```suggestion
```
